nixos/modules/config/shells-environment.nix

```diff
@@ -198,6 +206,11 @@ in
           # ~/bin if it exists overrides other bin directories.
           export PATH="$HOME/bin:$PATH"
         ''}
+
+        ${optionalString cfg.localBinInPath ''
+          # ~/.local/bin if it exists overrides other bin directories.
```

Comment:
that comment is not true. It is not actually checked if the directory exists.

Reply:
Bug in `homeBinInPath` as well then, will resolve based on convo wit h @ryantm

Reply:
Resolved in https://github.com/NixOS/nixpkgs/commit/55a211ae3184d105129f00da537c35e14197e300